src/Form/createForm.tsx

```diff
@@ -0,0 +1,300 @@
+      this.isKeyboardShowing = false;
+    };
+
+    setFieldInstance = (
+      index: number,
+      focusable: FocusableContainer | null
+    ) => {
+      this.instanceMap[index] = focusable;
+    };
+
+    focusNext = (index: number) => {
+      const indice = Object.keys(this.instanceMap)
+        .sort()
+        .map(Number)
+        .filter(i => i > index);
+      const i = indice[0];
+      if (i == null) {
+        Keyboard.dismiss();
+        return;
+      }
+      const focusableContainer = this.instanceMap[i];
+      if (
+        focusableContainer != null &&
+        focusableContainer.focusableRef != null &&
+        focusableContainer.focusableRef.current != null &&
+        focusableContainer.focusableRef.current.focus
+      ) {
+        InteractionManager.runAfterInteractions(() => {
+          const afterFocus = focusableContainer.focusableRef.current.focus();
+          if (afterFocus != null) {
+            if (afterFocus.dismissKeyboard) {
+              Keyboard.dismiss();
+              return;
+            }
+          }
+          if (this.isKeyboardShowing) {
+            InteractionManager.runAfterInteractions(() => {
+              this.scrollToFocusedTextInput();
+            });
+          }
+        });
+        return;
+      }
+      Keyboard.dismiss();
+    };
+
+    private clearAdjustContentOffsetTimeout() {
+      if (this.adjustContentOffsetTimeoutToken != null) {
+        clearTimeout(this.adjustContentOffsetTimeoutToken);
+        this.adjustContentOffsetTimeoutToken = null;
+      }
+    }
+
+    private scrollToFocusedTextInput() {
+      this.clearAdjustContentOffsetTimeout();
+      const currentlyFocusedNodeId = TextInput.State.currentlyFocusedField();
+      if (currentlyFocusedNodeId == null) {
+        return;
+      }
+      this.scrollToFocusedField(currentlyFocusedNodeId);
+    }
+
+    private scrollToFocusedField(nodeID: number) {
+      this.clearAdjustContentOffsetTimeout();
+
+      if (this.scrollViewRef.current == null) {
+        return;
+      }
+
+      const scrollViewNodeID = findNodeHandle(this.scrollViewRef.current);
+      if (scrollViewNodeID == null) {
+        return;
+      }
+
+      this.clearAdjustContentOffsetTimeout();
+
+      (UIManager as any).viewIsDescendantOf(
+        nodeID,
+        scrollViewNodeID,
+        (isAncestor: boolean) => {
+          if (!isAncestor) {
+            return;
+          }
+          UIManager.measureLayout(
+            nodeID,
+            scrollViewNodeID,
+            () => {},
+            (_x, y) => {
+              const { scrollViewContentHeight, scrollViewHeight } = this.state;
+              if (scrollViewContentHeight <= scrollViewHeight) {
+                return;
+              }
+              const maxContentOffsetY =
+                this.state.scrollViewContentHeight -
+                this.state.scrollViewHeight;
+              const desiredContentOffsetY = Math.min(
+                y < 100 ? 0 : (y * 2) / 3,
```

Comment:
Suppose we want to place the text input in the center of the scroll view, the content offset y we want should be `y + (H - h) / 2` where `H` is the scroll view height and `h` is the height of the text input.

Reply:
That will be weird ff the input positioned in centre of scrollview. 
Let's introduce a `scrollToInputThresholds` which does two things. 
1. If `textInput.y` is smaller the the thresholds, we will not scroll the scroll view. 
2. We will scroll to `textInput.y - scrollToInputThresholds`, which will simplify our math and life
Or, additionally we can have delegate for user calculating their desire scroll position

What do you think?

Reply:
Your suggestion would be enough. The name should be `scrollToTextInputOffset` though.

Reply:
We should have a default value for this new prop.